From 9eaf9bdbf7af731cabd62eb14a723432b73a337b Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@geldata.com>
Date: Thu, 31 Jul 2025 00:31:41 -0700
Subject: [PATCH] Make build executable suffix configurable

---
 configure    | 33 +++++++++++++++++++++++++++++++--
 configure.ac | 17 +++++++++++++++--
 2 files changed, 46 insertions(+), 4 deletions(-)

diff --git a/configure b/configure
index 483d2140a1c..7cc1746010f 100755
--- a/configure
+++ b/configure
@@ -1044,6 +1044,7 @@ with_emscripten_target
 enable_wasm_dynamic_linking
 enable_wasm_pthreads
 with_suffix
+with_build_suffix
 enable_shared
 with_static_libpython
 enable_profiling
@@ -1815,6 +1816,10 @@ Optional Packages:
                           Emscripten platform
   --with-suffix=SUFFIX    set executable suffix to SUFFIX (default is empty,
                           yes is mapped to '.exe')
+  --with-build-suffix=SUFFIX
+                          set build executable suffix to SUFFIX (default is
+                          value of --with-suffix, otheriwse '.exe' if building
+                          on a case-insensitive filesystem)
   --without-static-libpython
                           do not build libpythonMAJOR.MINOR.a and do not
                           install python.o (default is yes)
@@ -6606,6 +6611,31 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $EXEEXT" >&5
 $as_echo "$EXEEXT" >&6; }
 
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for --with-build-suffix" >&5
+$as_echo_n "checking for --with-build-suffix... " >&6; }
+
+# Check whether --with-build-suffix was given.
+if test "${with_build_suffix+set}" = set; then :
+  withval=$with_build_suffix;
+	case $with_build_suffix in #(
+  no) :
+    BUILDEXEEXT= ;; #(
+  yes) :
+    BUILDEXEEXT=.exe ;; #(
+  *) :
+    BUILDEXEEXT=$with_build_suffix
+   ;;
+esac
+
+else
+
+  BUILDEXEEXT=$EXEEXT
+
+fi
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $BUILDEXEEXT" >&5
+$as_echo "$BUILDEXEEXT" >&6; }
+
 # Test whether we're running on a non-case-sensitive system, in which
 # case we give a warning if no ext is given
 
@@ -6615,7 +6645,7 @@ if test ! -d CaseSensitiveTestDir; then
 mkdir CaseSensitiveTestDir
 fi
 
-if test -d casesensitivetestdir && test -z "$EXEEXT"
+if test -d casesensitivetestdir && test -z "$BUILDEXEEXT"
 then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
@@ -6623,7 +6653,6 @@ $as_echo "yes" >&6; }
 else
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
-	BUILDEXEEXT=$EXEEXT
 fi
 rmdir CaseSensitiveTestDir
 
diff --git a/configure.ac b/configure.ac
index 7ddc9970b38..d23c59354ab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1303,6 +1303,20 @@ AC_ARG_WITH([suffix],
 ])
 AC_MSG_RESULT([$EXEEXT])
 
+AC_MSG_CHECKING([for --with-build-suffix])
+AC_ARG_WITH([build-suffix],
+	    [AS_HELP_STRING([--with-build-suffix=SUFFIX], [set build executable suffix to SUFFIX (default is value of --with-suffix, otheriwse '.exe' if building on a case-insensitive filesystem)])],
+[
+	AS_CASE([$with_build_suffix],
+    [no], [BUILDEXEEXT=],
+    [yes], [BUILDEXEEXT=.exe],
+    [BUILDEXEEXT=$with_build_suffix]
+  )
+], [
+  BUILDEXEEXT=$EXEEXT
+])
+AC_MSG_RESULT([$BUILDEXEEXT])
+
 # Test whether we're running on a non-case-sensitive system, in which
 # case we give a warning if no ext is given
 AC_SUBST(BUILDEXEEXT)
@@ -1311,13 +1325,12 @@ if test ! -d CaseSensitiveTestDir; then
 mkdir CaseSensitiveTestDir
 fi
 
-if test -d casesensitivetestdir && test -z "$EXEEXT"
+if test -d casesensitivetestdir && test -z "$BUILDEXEEXT"
 then
     AC_MSG_RESULT(yes)
     BUILDEXEEXT=.exe
 else
 	AC_MSG_RESULT(no)
-	BUILDEXEEXT=$EXEEXT
 fi
 rmdir CaseSensitiveTestDir
 
-- 
2.49.1

